// @google/gemini-cli/bundle/dist-AXNLUOHP.js
const require = (await import('node:module')).createRequire(import.meta.url); const __chunk_filename = (await import('node:url')).fileURLToPath(import.meta.url); const __chunk_dirname = (await import('node:path')).dirname(__chunk_filename);
import {
  ADMIN_POLICY_TIER,
  ALLOWED_MCP_SERVER_PRIORITY,
  ALLOWED_TOOLS_FLAG_PRIORITY,
  ALWAYS_ALLOW_PRIORITY,
  AbortError,
  ActionStatus,
  ActivateSkillTool,
  ActivityDetector,
  ActivityMonitor,
  ActivityType,
  AdminControlsSettingsSchema,
  AfterAgentHookOutput,
  AfterModelHookOutput,
  AgentExecutionBlockedError,
  AgentExecutionStoppedError,
  AgentLoadError,
  AgentSession,
  AgentTerminateMode,
  ApiErrorEvent,
  ApiKeyUpdatedEvent,
  ApiRequestEvent,
  ApiRequestPhase,
  ApiResponseEvent,
  AskUserInvocation,
  AskUserTool,
  AuthProviderType,
  AuthType,
  BINARY_EXTENSIONS,
  BINARY_FILE_PATTERNS,
  BaseLlmClient,
  BeforeModelHookOutput,
  BeforeToolHookOutput,
  BeforeToolSelectionHookOutput,
  CODE_ASSIST_API_VERSION,
  CODE_ASSIST_ENDPOINT,
  COMMON_DIRECTORY_EXCLUDES,
  COMMON_IGNORE_PATTERNS,
  CacheService,
  ChatRecordingService,
  ClearcutLogger,
  CloseDiffRequestSchema,
  CodeAssistServer,
  ColorMode,
  CompressionStatus,
  Config,
  ConfigSchema,
  ConfigSource,
  ConsecaPolicyGenerationEvent,
  ConsecaVerdictEvent,
  ConversationFinishedEvent,
  ConversationInteractionInteraction,
  CoreToolCallStatus,
  CreditPurchaseClickEvent,
  CreditsUsedEvent,
  DEFAULT_CORE_POLICIES_DIR,
  DEFAULT_ENCODING,
  DEFAULT_FAST_ACK_MODEL_CONFIG_KEY,
  DEFAULT_FILE_EXCLUDES,
  DEFAULT_GUI_EDITOR,
  DEFAULT_MAX_ATTEMPTS,
  DEFAULT_MAX_INPUT_CHARS,
  DEFAULT_MAX_OUTPUT_CHARS,
  DEFAULT_MAX_TIME_MINUTES,
  DEFAULT_MAX_TURNS,
  DEFAULT_MODEL_CONFIGS,
  DEFAULT_OTLP_ENDPOINT,
  DEFAULT_POLICY_TIER,
  DEFAULT_QUERY_STRING,
  DEFAULT_TELEMETRY_TARGET,
  DEFAULT_TOKEN_LIMIT,
  DEFAULT_TRUNCATE_TOOL_OUTPUT_THRESHOLD,
  DefaultHookOutput,
  DiscoveredTool,
  EDITOR_DISPLAY_NAMES,
  EVENT_API_KEY_UPDATED,
  EVENT_CREDITS_USED,
  EVENT_CREDIT_PURCHASE_CLICK,
  EVENT_EMPTY_WALLET_MENU_SHOWN,
  EVENT_OVERAGE_MENU_SHOWN,
  EVENT_OVERAGE_OPTION_SELECTED,
  EXCLUDE_TOOLS_FLAG_PRIORITY,
  EXTENSION_POLICY_TIER,
  EditTool,
  EmptyWalletMenuShownEvent,
  EndSessionEvent,
  ExecutionLifecycleService,
  ExitCodes,
  ExperimentFlags,
  ExtensionDisableEvent,
  ExtensionEnableEvent,
  ExtensionInstallEvent,
  ExtensionIntegrityDataSchema,
  ExtensionIntegrityManager,
  ExtensionIntegrityMapSchema,
  ExtensionLoader,
  ExtensionUninstallEvent,
  ExtensionUpdateEvent,
  FETCH_FAILED_MESSAGE,
  FORCE_FILE_STORAGE_ENV_VAR,
  FRONTMATTER_REGEX,
  FetchAdminControlsResponseSchema,
  FetchError,
  FileDiscoveryService,
  FileExclusions,
  FileOperation,
  FileSchema,
  FileSearchFactory,
  FlashFallbackEvent,
  FolderTrustDiscoveryService,
  G1_CREDIT_TYPE,
  G1_UTM_CAMPAIGNS,
  GEMINI_CLI_COMPANION_EXTENSION_NAME,
  GEMINI_CLI_IDENTIFICATION_ENV_VAR,
  GEMINI_CLI_IDENTIFICATION_ENV_VAR_VALUE,
  GEN_AI_AGENT_DESCRIPTION,
  GEN_AI_AGENT_NAME,
  GEN_AI_CONVERSATION_ID,
  GEN_AI_INPUT_MESSAGES,
  GEN_AI_OPERATION_NAME,
  GEN_AI_OUTPUT_MESSAGES,
  GEN_AI_PROMPT_NAME,
  GEN_AI_REQUEST_MODEL,
  GEN_AI_RESPONSE_MODEL,
  GEN_AI_SYSTEM_INSTRUCTIONS,
  GEN_AI_TOOL_CALL_ID,
  GEN_AI_TOOL_DEFINITIONS,
  GEN_AI_TOOL_DESCRIPTION,
  GEN_AI_TOOL_NAME,
  GEN_AI_USAGE_INPUT_TOKENS,
  GEN_AI_USAGE_OUTPUT_TOKENS,
  GOVERNANCE_FILES,
  GcpLogExporter,
  GcpMetricExporter,
  GcpTraceExporter,
  GeminiChat,
  GeminiCliOperation,
  GeminiClient,
  GeminiEventType,
  GenAiOperationName,
  GenAiProviderName,
  GenAiTokenType,
  GitIgnoreParser,
  GitService,
  GlobTool,
  GrepTool,
  HOOKS_CONFIG_FIELDS,
  HighWaterMarkTracker,
  HookAggregator,
  HookEventHandler,
  HookEventName,
  HookPlanner,
  HookRegistry,
  HookRunner,
  HookSystem,
  HookType,
  IDEConnectionStatus,
  IDE_DEFINITIONS,
  IDE_MAX_OPEN_FILES,
  IDE_MAX_SELECTED_TEXT_LENGTH,
  IDE_REQUEST_TIMEOUT_MS,
  INCOMPLETE_JSON_MESSAGE,
  INITIAL_HISTORY_LENGTH,
  IdeClient,
  IdeConnectionEvent,
  IdeConnectionType,
  IdeContextNotificationSchema,
  IdeContextSchema,
  IdeContextStore,
  IdeDiffAcceptedNotificationSchema,
  IdeDiffClosedNotificationSchema,
  IdeDiffRejectedNotificationSchema,
  IneligibleTierError,
  IneligibleTierReasonCode,
  InitiationMethod,
  InjectionService,
  IntegrityDataStatus,
  IntegrityStatus,
  IntegrityStoreSchema,
  InvalidStreamError,
  JsonFormatter,
  JsonStreamEventType,
  KEYCHAIN_TEST_PREFIX,
  KeychainSchema,
  KeychainService,
  KeychainTokenStorage,
  LSTool,
  LegacyAgentProtocol,
  LegacyAgentSession,
  LlmRole,
  LocalAgentExecutor,
  LocalSandboxManager,
  Logger,
  LoggingContentGenerator,
  MCPDiscoveryState,
  MCPOAuthProvider,
  MCPOAuthTokenStorage,
  MCPServerConfig,
  MCPServerStatus,
  MCP_DEFAULT_TIMEOUT_MSEC,
  MCP_EXCLUDED_PRIORITY,
  MEDIA_FILE_PATTERNS,
  MIN_CREDIT_BALANCE,
  McpClient,
  McpConfigDefinitionSchema,
  MemoryContextManager,
  MemoryMetricType,
  MemoryMonitor,
  MessageBus,
  MessageSenderType,
  ModelConfigService,
  ModelNotFoundError,
  ModelRouterService,
  ModelSlashCommandEvent,
  NO_EDITOR_AVAILABLE_ERROR,
  NetworkRetryAttemptEvent,
  NoopSandboxManager,
  NotificationType,
  OAuthUtils,
  OUTPUT_UPDATE_INTERVAL_MS,
  OVERAGE_ELIGIBLE_MODELS,
  OnboardUserStatusCode,
  OnboardingStartEvent,
  OnboardingSuccessEvent,
  OpenDiffRequestSchema,
  OutputFormat,
  OverageMenuShownEvent,
  OverageOptionSelectedEvent,
  PYTHON_EXCLUDES,
  PerformanceMetricType,
  PlanErrorMessages,
  PolicyEngine,
  PolicyIntegrityManager,
  PreCompressTrigger,
  PrivateIpError,
  ProjectIdRequiredError,
  ROOT_SCHEDULER_ID,
  RateLimiter,
  ReadFileTool,
  ReadManyFilesTool,
  RecordingContentGenerator,
  ReleaseChannel,
  RequiredMcpServerConfigSchema,
  ResourceRegistry,
  RetryableQuotaError,
  RewindEvent,
  RipGrepTool,
  SCROLLBACK_LIMIT,
  SECRET_FILES,
  SECRET_PREFIX,
  SERVICE_DESCRIPTION,
  SERVICE_NAME,
  SESSION_FILE_PREFIX,
  SHELL_TOOL_NAMES,
  SUBAGENT_CANCELLED_ERROR_MESSAGE,
  SUBAGENT_REJECTED_ERROR_PREFIX,
  SYNTHETIC_THOUGHT_SIGNATURE,
  SYSTEM_FILE_EXCLUDES,
  SandboxedFileSystemService,
  Scheduler,
  SemanticAttributes,
  SessionEndReason,
  SessionStartSource,
  ShellExecutionService,
  ShellTool,
  ShellToolInvocation,
  SimpleExtensionLoader,
  SkillManager,
  SlashCommandStatus,
  StandardFileSystemService,
  StartSessionEvent,
  StartupProfiler,
  StreamEventType,
  StreamJsonFormatter,
  SubagentActivityErrorType,
  TASK_TYPE_LABELS,
  TOOL_OUTPUTS_DIR,
  TRUSTED_MCP_SERVER_PRIORITY,
  TaskStatus,
  TaskStatusSchema,
  TaskType,
  TaskTypeSchema,
  TelemetryTarget,
  TerminalQuotaError,
  ToolCallDecision,
  ToolCallEvent,
  ToolExecutionPhase,
  ToolExecutor,
  ToolOutputTruncatedEvent,
  ToolRegistry,
  TrackerAddDependencyTool,
  TrackerCreateTaskTool,
  TrackerGetTaskTool,
  TrackerListTasksTool,
  TrackerService,
  TrackerTaskSchema,
  TrackerUpdateTaskTool,
  TrackerVisualizeTool,
  Turn,
  UPGRADE_URL_PAGE,
  USER_POLICY_TIER,
  USER_STEERING_INSTRUCTION,
  UiTelemetryService,
  UserAccountManager,
  UserPromptEvent,
  UserTierId,
  ValidationCancelledError,
  ValidationRequiredError,
  WORKSPACE_POLICY_TIER,
  WarningPriority,
  WebFetchFallbackAttemptEvent,
  WebFetchTool,
  WebSearchTool,
  WindowsSandboxManager,
  WorkspaceContext,
  WorktreeService,
  WriteFileTool,
  WriteTodosTool,
  _clearCache,
  addMCPStatusChangeListener,
  addMemory,
  allowEditorTypeInSandbox,
  appendToLastTextPart,
  applyAdminAllowlist,
  applyReplacement,
  applyRequiredServers,
  assumeExhaustive,
  authEvents,
  buildG1Url,
  buildTodosReturnDisplay,
  buildToolResponseData,
  buildUserSteeringHintPrompt,
  bytesToMB,
  calculateReplacement,
  canUseRipgrep,
  checkExhaustive,
  checkPolicy,
  classifyGoogleError,
  cleanupWorktree,
  clearApiKey,
  clearCachedCredentialFile,
  clearEmittedPolicyWarnings,
  clearOauthClientCache,
  computeModelAddedAndRemovedLines,
  connectAndDiscover,
  connectToMcpServer,
  contentPartsToGeminiParts,
  convertColorToHex,
  convertGithubUrlToRaw,
  convertSessionToClientHistory,
  convertToFunctionResponse,
  convertToRestPayload,
  createCache,
  createCodeAssistContentGenerator,
  createContentGenerator,
  createContentGeneratorConfig,
  createConversationOffered,
  createHookOutput,
  createPolicyEngineConfig,
  createPolicyUpdater,
  createSafeProxyAgent,
  createSandboxManager,
  createSessionId,
  createTranslationState,
  createTransport,
  createUnauthorizedToolError,
  createWorkingStdio,
  createWorktree,
  createWorktreeService,
  decodeTagName,
  deleteSessionArtifactsAsync,
  deleteSubagentSessionDirAndArtifactsAsync,
  detectBOM,
  detectEncodingFromBuffer,
  detectFileType,
  detectIdeFromEnv,
  detectLineEnding,
  disableBracketedPasteMode,
  disableKittyKeyboardProtocol,
  disableLineWrapping,
  disableModifyOtherKeys,
  disableMouseEvents,
  disableMouseTracking,
  discoverMcpTools,
  discoverPrompts,
  discoverResources,
  discoverTools,
  doesToolInvocationMatch,
  enableBracketedPasteMode,
  enableKittyKeyboardProtocol,
  enableLineWrapping,
  enableModifyOtherKeys,
  enableMouseEvents,
  encodeTagName,
  ensureRgPath,
  enterAlternateScreen,
  escapeShellArg,
  estimateContextBreakdown,
  execStreaming,
  exitAlternateScreen,
  extractExtensionsFromPatterns,
  extractStringFromParseEntry,
  fetchAdminControls,
  fetchAdminControlsOnce,
  fetchWithTimeout,
  fileExists,
  filter,
  findGitRoot,
  findSecretFiles,
  flatMapTextParts,
  flattenMemory,
  flushTelemetry,
  formatBackgroundCompletionForModel,
  formatBytes,
  formatCheckpointDisplayList,
  formatForSpeech,
  formatPolicyError,
  formatProtoJsonDuration,
  formatTruncatedToolOutput,
  formatUserHintsForModel,
  geminiPartsToContentParts,
  generalistProfile,
  generateCheckpointFileName,
  generateFastAckText,
  generateSteeringAckMessage,
  generateSummary,
  getActivityDetector,
  getActivityMonitor,
  getAdminBlockedMcpServersMessage,
  getAdminErrorMessage,
  getAgentCardLoadOptions,
  getAllMCPServerStatuses,
  getAlwaysAllowPriorityFraction,
  getApprovalModeDescription,
  getAuthTypeFromEnv,
  getAvailablePort,
  getCachedEncodingForBuffer,
  getCheckpointInfoList,
  getCitations,
  getCodeAssistServer,
  getCommandName,
  getCommandRoot,
  getCommandRoots,
  getCompatibilityWarnings,
  getCompressionPrompt,
  getConsentForOauth,
  getConventionAttributes,
  getCoreSystemPrompt,
  getCorrectedFileContent,
  getDiffCommand,
  getDirectoryContextString,
  getEditorCommand,
  getEditorDisplayName,
  getEnvironmentContext,
  getErrorReplaceResult,
  getErrorStatus,
  getExperiments,
  getFileDiffFromResultDisplay,
  getFolderStructure,
  getFsErrorMessage,
  getFunctionCalls,
  getFunctionCallsAsJson,
  getFunctionCallsFromParts,
  getFunctionCallsFromPartsAsJson,
  getG1CreditBalance,
  getHookKey,
  getIdeInstaller,
  getInitialChatHistory,
  getMCPDiscoveryState,
  getMCPServerPrompts,
  getMCPServerStatus,
  getMemoryMonitor,
  getOauthClient,
  getPackageJson,
  getPathIdentity,
  getPlanModeExitMessage,
  getPolicyDenialError,
  getPolicyDirectories,
  getPolicyTier,
  getPreviousSession,
  getProjectRootForWorktree,
  getPromptIdWithFallback,
  getPty,
  getRealPath,
  getReleaseChannel,
  getRemoteAgentTargetUrl,
  getResponseText,
  getResponseTextFromParts,
  getRetryErrorType,
  getSecretFileFindArgs,
  getShellConfiguration,
  getSpecificMimeType,
  getStructuredResponse,
  getStructuredResponseFromParts,
  getSystemEncoding,
  getToolCallDataSchema,
  getToolSuggestion,
  getTruncatedCheckpointNames,
  getVersion,
  getWorktreePath,
  handleFallback,
  hasNetworkTransport,
  hasRedirection,
  hasValidEditorCommand,
  hasValidEditorCommandAsync,
  hasWorktreeChanges,
  ideContextStore,
  initializeActivityMonitor,
  initializeMemoryMonitor,
  initializeShellParsers,
  initializeTelemetry,
  invokeMcpPrompt,
  isAddressPrivate,
  isApiError,
  isAppleTerminal,
  isBinary,
  isBinaryFile,
  isCloudShell,
  isCompletedAskUserTool,
  isDumbTerminal,
  isEditToolParams,
  isEditorAvailable,
  isEditorAvailableAsync,
  isEmpty,
  isEnabled,
  isGeminiWorktree,
  isGitRepository,
  isGnuScreen,
  isGuiEditor,
  isHeadlessMode,
  isInvalidArgumentError,
  isJetBrainsTerminal,
  isLoopbackHost,
  isLowColorTmux,
  isNightly,
  isOverageEligibleModel,
  isPerformanceMonitoringActive,
  isPreview,
  isPrivateIp,
  isPrivateIpAsync,
  isRetryableError,
  isSchemaDepthError,
  isSecretFile,
  isStable,
  isStructuredError,
  isSubagentProgress,
  isTelemetrySdkInitialized,
  isTerminalEditor,
  isTmux,
  isToolActivityError,
  isToolCallResponseInfo,
  isUserActive,
  isUserVisibleHook,
  isValidNonThoughtTextPart,
  isWindows,
  isWindows10,
  isWithinRoot,
  isWriteFileToolParams,
  listExtensions,
  listMemoryFiles,
  loadAgentsFromDirectory,
  loadApiKey,
  loadExtensionPolicies,
  loadPoliciesFromToml,
  loadSkillFromFile,
  loadSkillsFromDir,
  loadWasmBinary,
  logApiError,
  logApiRequest,
  logApiResponse,
  logBillingEvent,
  logChatCompression,
  logCliConfiguration,
  logConsecaPolicyGeneration,
  logConsecaVerdict,
  logConversationFinishedEvent,
  logExtensionDisable,
  logExtensionEnable,
  logExtensionInstallEvent,
  logExtensionUninstall,
  logExtensionUpdateEvent,
  logFlashFallback,
  logIdeConnection,
  logModelSlashCommand,
  logNetworkRetryAttempt,
  logOnboardingStart,
  logOnboardingSuccess,
  logRewind,
  logSlashCommand,
  logToolCall,
  logToolOutputTruncated,
  logUserPrompt,
  logWebFetchFallbackAttempt,
  makeChatCompressionEvent,
  makeFakeConfig,
  makeSlashCommandEvent,
  mapError,
  mapFinishReason,
  mapHttpToGrpcStatus,
  mapUsage,
  markdownToAgentDefinition,
  mcpServerRequiresOAuth,
  normalizeCommand,
  normalizeSpace,
  normalizeUrl,
  openBrowserSecurely,
  openDiff,
  parseAgentMarkdown,
  parseAndFormatApiError,
  parseBashCommandDetails,
  parseBooleanEnvFlag,
  parseCommandDetails,
  parseFrontmatter,
  parsePrompt,
  parseTelemetryTargetValue,
  parseThought,
  partListUnionToString,
  partToString,
  patchStdio,
  performInit,
  performRestore,
  populateMcpServerCommand,
  processRestorableToolCalls,
  processSingleFileContent,
  promptIdContext,
  readFileWithEncoding,
  readPathFromWorkspace,
  readPolicyFiles,
  readWasmBinaryFromDisk,
  recordApiErrorMetrics,
  recordApiRequestBreakdown,
  recordApiResponseMetrics,
  recordBaselineComparison,
  recordContentRetry,
  recordContentRetryFailure,
  recordConversationOffered,
  recordCpuUsage,
  recordCreditPurchaseClick,
  recordCurrentMemoryUsage,
  recordCustomApiResponseMetrics,
  recordCustomTokenUsageMetrics,
  recordExitFail,
  recordFileOperationMetric,
  recordFlickerFrame,
  recordGenAiClientOperationDuration,
  recordGenAiClientTokenUsage,
  recordInvalidChunk,
  recordMemoryUsage,
  recordModelRoutingMetrics,
  recordOverageOptionSelected,
  recordPerformanceRegression,
  recordPerformanceScore,
  recordRetryAttemptMetrics,
  recordSlowRender,
  recordStartupPerformance,
  recordTokenEfficiency,
  recordTokenUsageMetrics,
  recordToolCallInteractions,
  recordToolCallMetrics,
  recordToolExecutionBreakdown,
  recordToolQueueDepth,
  recordUserActivity,
  refreshMemory,
  removeMCPStatusChangeListener,
  resetApiKeyCacheForTesting,
  resetBrowserSession,
  resetEncodingCache,
  resetOauthClientForTesting,
  resetUserDataCacheForTesting,
  resetVersionCache,
  resolveEditorAsync,
  resolveExecutable,
  resolvePathFromEnv,
  resolveSandboxPaths,
  resolveTelemetrySettings,
  retryWithBackoff,
  runInDevTraceSpan,
  safeLiteralReplace,
  safeTemplateReplace,
  sanitizeAdminSettings,
  sanitizeFilenamePart,
  sanitizeHostname,
  sanitizeOutput,
  sanitizePaths,
  saveApiKey,
  saveTruncatedToolOutput,
  scheduleAgentTools,
  serializeTerminalToObject,
  sessionId,
  setGlobalProxy,
  setupUser,
  shouldAttemptBrowserLaunch,
  shouldAutoUseCredits,
  shouldEnterAlternateScreen,
  shouldHideToolCall,
  shouldLaunchBrowser,
  shouldShowEmptyWalletMenu,
  shouldShowOverageMenu,
  showMemory,
  shutdownTelemetry,
  sortFileEntries,
  spawnAsync,
  splitCommands,
  startGlobalActivityMonitoring,
  startGlobalMemoryMonitoring,
  startMemoryService,
  startupProfiler,
  stopAdminControlsPolling,
  stopGlobalActivityMonitoring,
  stopGlobalMemoryMonitoring,
  stripShellWrapper,
  supports256Colors,
  supportsTrueColor,
  tokenLimit,
  toolResultDisplayToContentParts,
  translateEvent,
  truncateFastAckInput,
  truncateString,
  tryRealpath,
  uiTelemetryService,
  updateGlobalFetchTimeouts,
  updateMCPServerStatus,
  updatePolicy,
  validateAndSanitizeSessionId,
  validateMcpPolicyToolNames,
  validatePlanContent,
  validatePlanPath,
  windowsCodePageToEncoding,
  wrapInAccountChooser,
  writeToStderr,
  writeToStdout
} from "./chunk-GOUPAQ35.js";
import {
  ACTIVATE_SKILL_TOOL_NAME,
  ALL_BUILTIN_TOOL_NAMES,
  ALWAYS_ALLOW_PRIORITY_FRACTION,
  ALWAYS_ALLOW_PRIORITY_OFFSET,
  ASK_USER_DISPLAY_NAME,
  ASK_USER_OPTION_PARAM_DESCRIPTION,
  ASK_USER_OPTION_PARAM_LABEL,
  ASK_USER_PARAM_QUESTIONS,
  ASK_USER_QUESTION_PARAM_HEADER,
  ASK_USER_QUESTION_PARAM_MULTI_SELECT,
  ASK_USER_QUESTION_PARAM_OPTIONS,
  ASK_USER_QUESTION_PARAM_PLACEHOLDER,
  ASK_USER_QUESTION_PARAM_QUESTION,
  ASK_USER_QUESTION_PARAM_TYPE,
  ASK_USER_TOOL_NAME,
  AccountSuspendedError,
  ApprovalMode,
  BadRequestError,
  BaseDeclarativeTool,
  BaseToolInvocation,
  COMPLETE_TASK_DISPLAY_NAME,
  COMPLETE_TASK_TOOL_NAME,
  CanceledError,
  ChangeAuthRequestedError,
  CoreEvent,
  CoreEventEmitter,
  DEFAULT_CONTEXT_FILENAME,
  DEFAULT_FILE_FILTERING_OPTIONS,
  DEFAULT_GEMINI_EMBEDDING_MODEL,
  DEFAULT_GEMINI_FLASH_LITE_MODEL,
  DEFAULT_GEMINI_FLASH_MODEL,
  DEFAULT_GEMINI_MODEL,
  DEFAULT_GEMINI_MODEL_AUTO,
  DEFAULT_MAX_LINES_TEXT_FILE,
  DEFAULT_MEMORY_FILE_FILTERING_OPTIONS,
  DEFAULT_THINKING_MODE,
  DISCOVERED_TOOL_PREFIX,
  DOCS_PARAM_PATH,
  DeclarativeTool,
  DiscoveredMCPTool,
  DiscoveredMCPToolInvocation,
  EDIT_DISPLAY_NAME,
  EDIT_PARAM_ALLOW_MULTIPLE,
  EDIT_PARAM_INSTRUCTION,
  EDIT_PARAM_NEW_STRING,
  EDIT_PARAM_OLD_STRING,
  EDIT_TOOL_NAME,
  EDIT_TOOL_NAMES,
  ENTER_PLAN_MODE_TOOL_NAME,
  EXIT_PLAN_MODE_TOOL_NAME,
  EXIT_PLAN_PARAM_PLAN_FILENAME,
  FatalAuthenticationError,
  FatalCancellationError,
  FatalConfigError,
  FatalError,
  FatalInputError,
  FatalSandboxError,
  FatalToolExecutionError,
  FatalTurnLimitedError,
  ForbiddenError,
  GEMINI_DIR,
  GEMINI_IGNORE_FILE_NAME,
  GEMINI_MODEL_ALIAS_AUTO,
  GEMINI_MODEL_ALIAS_FLASH,
  GEMINI_MODEL_ALIAS_FLASH_LITE,
  GEMINI_MODEL_ALIAS_PRO,
  GET_INTERNAL_DOCS_TOOL_NAME,
  GLOB_DISPLAY_NAME,
  GLOB_TOOL_NAME,
  GOOGLE_ACCOUNTS_FILENAME,
  GREP_DISPLAY_NAME,
  GREP_PARAM_AFTER,
  GREP_PARAM_BEFORE,
  GREP_PARAM_CONTEXT,
  GREP_PARAM_EXCLUDE_PATTERN,
  GREP_PARAM_FIXED_STRINGS,
  GREP_PARAM_INCLUDE_PATTERN,
  GREP_PARAM_MAX_MATCHES_PER_FILE,
  GREP_PARAM_NAMES_ONLY,
  GREP_PARAM_NO_IGNORE,
  GREP_PARAM_TOTAL_MAX_MATCHES,
  GREP_TOOL_NAME,
  INTEGRITY_FILENAME,
  INTEGRITY_KEY_FILENAME,
  InProcessCheckerType,
  KEYCHAIN_SERVICE_NAME,
  Kind,
  LS_DISPLAY_NAME,
  LS_PARAM_IGNORE,
  LS_TOOL_NAME,
  MAX_FILE_SIZE_MB,
  MAX_LINE_LENGTH_TEXT_FILE,
  MCP_QUALIFIED_NAME_SEPARATOR,
  MCP_TOOL_PREFIX,
  MEMORY_PARAM_FACT,
  MEMORY_SECTION_HEADER,
  MEMORY_TOOL_NAME,
  MODES_BY_PERMISSIVENESS,
  MUTATOR_KINDS,
  MemoryTool,
  MessageBusType,
  PARAM_CASE_SENSITIVE,
  PARAM_DESCRIPTION,
  PARAM_DIR_PATH,
  PARAM_FILE_FILTERING_OPTIONS,
  PARAM_FILE_PATH,
  PARAM_PATTERN,
  PARAM_RESPECT_GEMINI_IGNORE,
  PARAM_RESPECT_GIT_IGNORE,
  PLAN_MODE_PARAM_REASON,
  PLAN_MODE_TOOLS,
  PREVIEW_GEMINI_3_1_CUSTOM_TOOLS_MODEL,
  PREVIEW_GEMINI_3_1_FLASH_LITE_MODEL,
  PREVIEW_GEMINI_3_1_MODEL,
  PREVIEW_GEMINI_FLASH_MODEL,
  PREVIEW_GEMINI_MODEL,
  PREVIEW_GEMINI_MODEL_AUTO,
  PRIORITY_SUBAGENT_TOOL,
  PRIORITY_YOLO_ALLOW_ALL,
  PolicyDecision,
  QuestionType,
  READ_FILE_DISPLAY_NAME,
  READ_FILE_PARAM_END_LINE,
  READ_FILE_PARAM_START_LINE,
  READ_FILE_TOOL_NAME,
  READ_MANY_FILES_DISPLAY_NAME,
  READ_MANY_FILES_TOOL_NAME,
  READ_MANY_PARAM_EXCLUDE,
  READ_MANY_PARAM_INCLUDE,
  READ_MANY_PARAM_RECURSIVE,
  READ_MANY_PARAM_USE_DEFAULT_EXCLUDES,
  READ_ONLY_KINDS,
  REFERENCE_CONTENT_END,
  REFERENCE_CONTENT_START,
  SECRET_KEY_ACCOUNT,
  SHELL_PARAM_COMMAND,
  SHELL_PARAM_IS_BACKGROUND,
  SHELL_TOOL_NAME,
  SKILL_PARAM_NAME,
  SchemaValidator,
  Storage,
  TODOS_ITEM_PARAM_DESCRIPTION,
  TODOS_ITEM_PARAM_STATUS,
  TODOS_PARAM_TODOS,
  TOOLS_REQUIRING_NARROWING,
  TOOL_LEGACY_ALIASES,
  TOPIC_PARAM_STRATEGIC_INTENT,
  TOPIC_PARAM_SUMMARY,
  TOPIC_PARAM_TITLE,
  TRACKER_ADD_DEPENDENCY_TOOL_NAME,
  TRACKER_CREATE_TASK_TOOL_NAME,
  TRACKER_GET_TASK_TOOL_NAME,
  TRACKER_LIST_TASKS_TOOL_NAME,
  TRACKER_UPDATE_TASK_TOOL_NAME,
  TRACKER_VISUALIZE_TOOL_NAME,
  ToolConfirmationOutcome,
  ToolErrorType,
  UPDATE_TOPIC_DISPLAY_NAME,
  UPDATE_TOPIC_TOOL_NAME,
  UnauthorizedError,
  VALID_GEMINI_MODELS,
  WEB_FETCH_DISPLAY_NAME,
  WEB_FETCH_PARAM_PROMPT,
  WEB_FETCH_TOOL_NAME,
  WEB_SEARCH_DISPLAY_NAME,
  WEB_SEARCH_PARAM_QUERY,
  WEB_SEARCH_TOOL_NAME,
  WRITE_FILE_DISPLAY_NAME,
  WRITE_FILE_PARAM_CONTENT,
  WRITE_FILE_TOOL_NAME,
  WRITE_TODOS_TOOL_NAME,
  assertValidPathString,
  categorizeAndConcatenate,
  concatenateInstructions,
  coreEvents,
  debugLogger,
  deduplicatePathsByFileIdentity,
  escapePath,
  formatMcpToolName,
  generateValidName,
  getAllGeminiMdFilenames,
  getCurrentGeminiMdFilename,
  getDisplayString,
  getEnvironmentMemoryPaths,
  getErrorMessage,
  getErrorType,
  getExtensionMemoryPaths,
  getGlobalMemoryFilePath,
  getGlobalMemoryPaths,
  getHookSource,
  getProjectHash,
  getProjectMemoryFilePath,
  getToolAliases,
  getUserProjectMemoryPaths,
  hasCycleInSchema,
  hasSummary,
  homedir,
  isAbortError,
  isAccountSuspendedError,
  isActiveModel,
  isAuthenticationError,
  isAutoModel,
  isBackgroundExecutionData,
  isCustomModel,
  isFatalToolError,
  isFileDiff,
  isGemini2Model,
  isGemini3Model,
  isGrepResult,
  isListResult,
  isMcpToolAnnotation,
  isMcpToolName,
  isNodeError,
  isPreviewModel,
  isProModel,
  isReadManyFilesResult,
  isRecord,
  isStructuredToolResult,
  isSubpath,
  isTool,
  isValidPathString,
  isValidToolName,
  jsonToMarkdown,
  loadJitSubdirectoryMemory,
  loadServerHierarchicalMemory,
  makeRelative,
  normalizePath,
  parseGoogleApiError,
  parseMcpToolName,
  readGeminiMdFiles,
  refreshServerHierarchicalMemory,
  resolveClassifierModel,
  resolveModel,
  resolveToRealPath,
  safeJsonToMarkdown,
  setGeminiMdFilename,
  shortenPath,
  supportsModernFeatures,
  supportsMultimodalFunctionResponse,
  tildeifyPath,
  tmpdir,
  toFriendlyError,
  unescapePath
} from "./chunk-ETUADTWF.js";
import "./chunk-664ZODQF.js";
import "./chunk-RJTRUG2J.js";
import {
  SpanStatusCode,
  ValueType
} from "./chunk-IUUIT4SU.js";
import "./chunk-34MYV7JD.js";
export {
  ACTIVATE_SKILL_TOOL_NAME,
  ADMIN_POLICY_TIER,
  ALLOWED_MCP_SERVER_PRIORITY,
  ALLOWED_TOOLS_FLAG_PRIORITY,
  ALL_BUILTIN_TOOL_NAMES,
  ALWAYS_ALLOW_PRIORITY,
  ALWAYS_ALLOW_PRIORITY_FRACTION,
  ALWAYS_ALLOW_PRIORITY_OFFSET,
  ASK_USER_DISPLAY_NAME,
  ASK_USER_OPTION_PARAM_DESCRIPTION,
  ASK_USER_OPTION_PARAM_LABEL,
  ASK_USER_PARAM_QUESTIONS,
  ASK_USER_QUESTION_PARAM_HEADER,
  ASK_USER_QUESTION_PARAM_MULTI_SELECT,
  ASK_USER_QUESTION_PARAM_OPTIONS,
  ASK_USER_QUESTION_PARAM_PLACEHOLDER,
  ASK_USER_QUESTION_PARAM_QUESTION,
  ASK_USER_QUESTION_PARAM_TYPE,
  ASK_USER_TOOL_NAME,
  AbortError,
  AccountSuspendedError,
  ActionStatus,
  ActivateSkillTool,
  ActivityDetector,
  ActivityMonitor,
  ActivityType,
  AdminControlsSettingsSchema,
  AfterAgentHookOutput,
  AfterModelHookOutput,
  AgentExecutionBlockedError,
  AgentExecutionStoppedError,
  AgentLoadError,
  AgentSession,
  AgentTerminateMode,
  ApiErrorEvent,
  ApiKeyUpdatedEvent,
  ApiRequestEvent,
  ApiRequestPhase,
  ApiResponseEvent,
  ApprovalMode,
  AskUserInvocation,
  AskUserTool,
  AuthProviderType,
  AuthType,
  BINARY_EXTENSIONS,
  BINARY_FILE_PATTERNS,
  BadRequestError,
  BaseDeclarativeTool,
  BaseLlmClient,
  BaseToolInvocation,
  BeforeModelHookOutput,
  BeforeToolHookOutput,
  BeforeToolSelectionHookOutput,
  CODE_ASSIST_API_VERSION,
  CODE_ASSIST_ENDPOINT,
  COMMON_DIRECTORY_EXCLUDES,
  COMMON_IGNORE_PATTERNS,
  COMPLETE_TASK_DISPLAY_NAME,
  COMPLETE_TASK_TOOL_NAME,
  CacheService,
  CanceledError,
  ChangeAuthRequestedError,
  ChatRecordingService,
  ClearcutLogger,
  CloseDiffRequestSchema,
  CodeAssistServer,
  ColorMode,
  CompressionStatus,
  Config,
  ConfigSchema,
  ConfigSource,
  ConsecaPolicyGenerationEvent,
  ConsecaVerdictEvent,
  ConversationFinishedEvent,
  ConversationInteractionInteraction,
  CoreEvent,
  CoreEventEmitter,
  CoreToolCallStatus,
  CreditPurchaseClickEvent,
  CreditsUsedEvent,
  DEFAULT_CONTEXT_FILENAME,
  DEFAULT_CORE_POLICIES_DIR,
  DEFAULT_ENCODING,
  DEFAULT_FAST_ACK_MODEL_CONFIG_KEY,
  DEFAULT_FILE_EXCLUDES,
  DEFAULT_FILE_FILTERING_OPTIONS,
  DEFAULT_GEMINI_EMBEDDING_MODEL,
  DEFAULT_GEMINI_FLASH_LITE_MODEL,
  DEFAULT_GEMINI_FLASH_MODEL,
  DEFAULT_GEMINI_MODEL,
  DEFAULT_GEMINI_MODEL_AUTO,
  DEFAULT_GUI_EDITOR,
  DEFAULT_MAX_ATTEMPTS,
  DEFAULT_MAX_INPUT_CHARS,
  DEFAULT_MAX_LINES_TEXT_FILE,
  DEFAULT_MAX_OUTPUT_CHARS,
  DEFAULT_MAX_TIME_MINUTES,
  DEFAULT_MAX_TURNS,
  DEFAULT_MEMORY_FILE_FILTERING_OPTIONS,
  DEFAULT_MODEL_CONFIGS,
  DEFAULT_OTLP_ENDPOINT,
  DEFAULT_POLICY_TIER,
  DEFAULT_QUERY_STRING,
  DEFAULT_TELEMETRY_TARGET,
  DEFAULT_THINKING_MODE,
  DEFAULT_TOKEN_LIMIT,
  DEFAULT_TRUNCATE_TOOL_OUTPUT_THRESHOLD,
  DISCOVERED_TOOL_PREFIX,
  DOCS_PARAM_PATH,
  DeclarativeTool,
  DefaultHookOutput,
  DiscoveredMCPTool,
  DiscoveredMCPToolInvocation,
  DiscoveredTool,
  EDITOR_DISPLAY_NAMES,
  EDIT_DISPLAY_NAME,
  EDIT_PARAM_ALLOW_MULTIPLE,
  EDIT_PARAM_INSTRUCTION,
  EDIT_PARAM_NEW_STRING,
  EDIT_PARAM_OLD_STRING,
  EDIT_TOOL_NAME,
  EDIT_TOOL_NAMES,
  ENTER_PLAN_MODE_TOOL_NAME,
  EVENT_API_KEY_UPDATED,
  EVENT_CREDITS_USED,
  EVENT_CREDIT_PURCHASE_CLICK,
  EVENT_EMPTY_WALLET_MENU_SHOWN,
  EVENT_OVERAGE_MENU_SHOWN,
  EVENT_OVERAGE_OPTION_SELECTED,
  EXCLUDE_TOOLS_FLAG_PRIORITY,
  EXIT_PLAN_MODE_TOOL_NAME,
  EXIT_PLAN_PARAM_PLAN_FILENAME,
  EXTENSION_POLICY_TIER,
  EditTool,
  EmptyWalletMenuShownEvent,
  EndSessionEvent,
  ExecutionLifecycleService,
  ExitCodes,
  ExperimentFlags,
  ExtensionDisableEvent,
  ExtensionEnableEvent,
  ExtensionInstallEvent,
  ExtensionIntegrityDataSchema,
  ExtensionIntegrityManager,
  ExtensionIntegrityMapSchema,
  ExtensionLoader,
  ExtensionUninstallEvent,
  ExtensionUpdateEvent,
  FETCH_FAILED_MESSAGE,
  FORCE_FILE_STORAGE_ENV_VAR,
  FRONTMATTER_REGEX,
  FatalAuthenticationError,
  FatalCancellationError,
  FatalConfigError,
  FatalError,
  FatalInputError,
  FatalSandboxError,
  FatalToolExecutionError,
  FatalTurnLimitedError,
  FetchAdminControlsResponseSchema,
  FetchError,
  FileDiscoveryService,
  FileExclusions,
  FileOperation,
  FileSchema,
  FileSearchFactory,
  FlashFallbackEvent,
  FolderTrustDiscoveryService,
  ForbiddenError,
  G1_CREDIT_TYPE,
  G1_UTM_CAMPAIGNS,
  GEMINI_CLI_COMPANION_EXTENSION_NAME,
  GEMINI_CLI_IDENTIFICATION_ENV_VAR,
  GEMINI_CLI_IDENTIFICATION_ENV_VAR_VALUE,
  GEMINI_DIR,
  GEMINI_IGNORE_FILE_NAME,
  GEMINI_MODEL_ALIAS_AUTO,
  GEMINI_MODEL_ALIAS_FLASH,
  GEMINI_MODEL_ALIAS_FLASH_LITE,
  GEMINI_MODEL_ALIAS_PRO,
  GEN_AI_AGENT_DESCRIPTION,
  GEN_AI_AGENT_NAME,
  GEN_AI_CONVERSATION_ID,
  GEN_AI_INPUT_MESSAGES,
  GEN_AI_OPERATION_NAME,
  GEN_AI_OUTPUT_MESSAGES,
  GEN_AI_PROMPT_NAME,
  GEN_AI_REQUEST_MODEL,
  GEN_AI_RESPONSE_MODEL,
  GEN_AI_SYSTEM_INSTRUCTIONS,
  GEN_AI_TOOL_CALL_ID,
  GEN_AI_TOOL_DEFINITIONS,
  GEN_AI_TOOL_DESCRIPTION,
  GEN_AI_TOOL_NAME,
  GEN_AI_USAGE_INPUT_TOKENS,
  GEN_AI_USAGE_OUTPUT_TOKENS,
  GET_INTERNAL_DOCS_TOOL_NAME,
  GLOB_DISPLAY_NAME,
  GLOB_TOOL_NAME,
  GOOGLE_ACCOUNTS_FILENAME,
  GOVERNANCE_FILES,
  GREP_DISPLAY_NAME,
  GREP_PARAM_AFTER,
  GREP_PARAM_BEFORE,
  GREP_PARAM_CONTEXT,
  GREP_PARAM_EXCLUDE_PATTERN,
  GREP_PARAM_FIXED_STRINGS,
  GREP_PARAM_INCLUDE_PATTERN,
  GREP_PARAM_MAX_MATCHES_PER_FILE,
  GREP_PARAM_NAMES_ONLY,
  GREP_PARAM_NO_IGNORE,
  GREP_PARAM_TOTAL_MAX_MATCHES,
  GREP_TOOL_NAME,
  GcpLogExporter,
  GcpMetricExporter,
  GcpTraceExporter,
  GeminiChat,
  GeminiCliOperation,
  GeminiClient,
  GeminiEventType,
  GenAiOperationName,
  GenAiProviderName,
  GenAiTokenType,
  GitIgnoreParser,
  GitService,
  GlobTool,
  GrepTool,
  HOOKS_CONFIG_FIELDS,
  HighWaterMarkTracker,
  HookAggregator,
  HookEventHandler,
  HookEventName,
  HookPlanner,
  HookRegistry,
  HookRunner,
  HookSystem,
  HookType,
  IDEConnectionStatus,
  IDE_DEFINITIONS,
  IDE_MAX_OPEN_FILES,
  IDE_MAX_SELECTED_TEXT_LENGTH,
  IDE_REQUEST_TIMEOUT_MS,
  INCOMPLETE_JSON_MESSAGE,
  INITIAL_HISTORY_LENGTH,
  INTEGRITY_FILENAME,
  INTEGRITY_KEY_FILENAME,
  IdeClient,
  IdeConnectionEvent,
  IdeConnectionType,
  IdeContextNotificationSchema,
  IdeContextSchema,
  IdeContextStore,
  IdeDiffAcceptedNotificationSchema,
  IdeDiffClosedNotificationSchema,
  IdeDiffRejectedNotificationSchema,
  InProcessCheckerType,
  IneligibleTierError,
  IneligibleTierReasonCode,
  InitiationMethod,
  InjectionService,
  IntegrityDataStatus,
  IntegrityStatus,
  IntegrityStoreSchema,
  InvalidStreamError,
  JsonFormatter,
  JsonStreamEventType,
  KEYCHAIN_SERVICE_NAME,
  KEYCHAIN_TEST_PREFIX,
  KeychainSchema,
  KeychainService,
  KeychainTokenStorage,
  Kind,
  LSTool,
  LS_DISPLAY_NAME,
  LS_PARAM_IGNORE,
  LS_TOOL_NAME,
  LegacyAgentProtocol,
  LegacyAgentSession,
  LlmRole,
  LocalAgentExecutor,
  LocalSandboxManager,
  Logger,
  LoggingContentGenerator,
  MAX_FILE_SIZE_MB,
  MAX_LINE_LENGTH_TEXT_FILE,
  MCPDiscoveryState,
  MCPOAuthProvider,
  MCPOAuthTokenStorage,
  MCPServerConfig,
  MCPServerStatus,
  MCP_DEFAULT_TIMEOUT_MSEC,
  MCP_EXCLUDED_PRIORITY,
  MCP_QUALIFIED_NAME_SEPARATOR,
  MCP_TOOL_PREFIX,
  MEDIA_FILE_PATTERNS,
  MEMORY_PARAM_FACT,
  MEMORY_SECTION_HEADER,
  MEMORY_TOOL_NAME,
  MIN_CREDIT_BALANCE,
  MODES_BY_PERMISSIVENESS,
  MUTATOR_KINDS,
  McpClient,
  McpConfigDefinitionSchema,
  MemoryContextManager,
  MemoryMetricType,
  MemoryMonitor,
  MemoryTool,
  MessageBus,
  MessageBusType,
  MessageSenderType,
  ModelConfigService,
  ModelNotFoundError,
  ModelRouterService,
  ModelSlashCommandEvent,
  NO_EDITOR_AVAILABLE_ERROR,
  NetworkRetryAttemptEvent,
  NoopSandboxManager,
  NotificationType,
  OAuthUtils,
  OUTPUT_UPDATE_INTERVAL_MS,
  OVERAGE_ELIGIBLE_MODELS,
  OnboardUserStatusCode,
  OnboardingStartEvent,
  OnboardingSuccessEvent,
  OpenDiffRequestSchema,
  OutputFormat,
  OverageMenuShownEvent,
  OverageOptionSelectedEvent,
  PARAM_CASE_SENSITIVE,
  PARAM_DESCRIPTION,
  PARAM_DIR_PATH,
  PARAM_FILE_FILTERING_OPTIONS,
  PARAM_FILE_PATH,
  PARAM_PATTERN,
  PARAM_RESPECT_GEMINI_IGNORE,
  PARAM_RESPECT_GIT_IGNORE,
  PLAN_MODE_PARAM_REASON,
  PLAN_MODE_TOOLS,
  PREVIEW_GEMINI_3_1_CUSTOM_TOOLS_MODEL,
  PREVIEW_GEMINI_3_1_FLASH_LITE_MODEL,
  PREVIEW_GEMINI_3_1_MODEL,
  PREVIEW_GEMINI_FLASH_MODEL,
  PREVIEW_GEMINI_MODEL,
  PREVIEW_GEMINI_MODEL_AUTO,
  PRIORITY_SUBAGENT_TOOL,
  PRIORITY_YOLO_ALLOW_ALL,
  PYTHON_EXCLUDES,
  PerformanceMetricType,
  PlanErrorMessages,
  PolicyDecision,
  PolicyEngine,
  PolicyIntegrityManager,
  PreCompressTrigger,
  PrivateIpError,
  ProjectIdRequiredError,
  QuestionType,
  READ_FILE_DISPLAY_NAME,
  READ_FILE_PARAM_END_LINE,
  READ_FILE_PARAM_START_LINE,
  READ_FILE_TOOL_NAME,
  READ_MANY_FILES_DISPLAY_NAME,
  READ_MANY_FILES_TOOL_NAME,
  READ_MANY_PARAM_EXCLUDE,
  READ_MANY_PARAM_INCLUDE,
  READ_MANY_PARAM_RECURSIVE,
  READ_MANY_PARAM_USE_DEFAULT_EXCLUDES,
  READ_ONLY_KINDS,
  REFERENCE_CONTENT_END,
  REFERENCE_CONTENT_START,
  ROOT_SCHEDULER_ID,
  RateLimiter,
  ReadFileTool,
  ReadManyFilesTool,
  RecordingContentGenerator,
  ReleaseChannel,
  RequiredMcpServerConfigSchema,
  ResourceRegistry,
  RetryableQuotaError,
  RewindEvent,
  RipGrepTool,
  SCROLLBACK_LIMIT,
  SECRET_FILES,
  SECRET_KEY_ACCOUNT,
  SECRET_PREFIX,
  SERVICE_DESCRIPTION,
  SERVICE_NAME,
  SESSION_FILE_PREFIX,
  SHELL_PARAM_COMMAND,
  SHELL_PARAM_IS_BACKGROUND,
  SHELL_TOOL_NAME,
  SHELL_TOOL_NAMES,
  SKILL_PARAM_NAME,
  SUBAGENT_CANCELLED_ERROR_MESSAGE,
  SUBAGENT_REJECTED_ERROR_PREFIX,
  SYNTHETIC_THOUGHT_SIGNATURE,
  SYSTEM_FILE_EXCLUDES,
  SandboxedFileSystemService,
  Scheduler,
  SchemaValidator,
  SemanticAttributes,
  SessionEndReason,
  SessionStartSource,
  ShellExecutionService,
  ShellTool,
  ShellToolInvocation,
  SimpleExtensionLoader,
  SkillManager,
  SlashCommandStatus,
  SpanStatusCode,
  StandardFileSystemService,
  StartSessionEvent,
  StartupProfiler,
  Storage,
  StreamEventType,
  StreamJsonFormatter,
  SubagentActivityErrorType,
  TASK_TYPE_LABELS,
  TODOS_ITEM_PARAM_DESCRIPTION,
  TODOS_ITEM_PARAM_STATUS,
  TODOS_PARAM_TODOS,
  TOOLS_REQUIRING_NARROWING,
  TOOL_LEGACY_ALIASES,
  TOOL_OUTPUTS_DIR,
  TOPIC_PARAM_STRATEGIC_INTENT,
  TOPIC_PARAM_SUMMARY,
  TOPIC_PARAM_TITLE,
  TRACKER_ADD_DEPENDENCY_TOOL_NAME,
  TRACKER_CREATE_TASK_TOOL_NAME,
  TRACKER_GET_TASK_TOOL_NAME,
  TRACKER_LIST_TASKS_TOOL_NAME,
  TRACKER_UPDATE_TASK_TOOL_NAME,
  TRACKER_VISUALIZE_TOOL_NAME,
  TRUSTED_MCP_SERVER_PRIORITY,
  TaskStatus,
  TaskStatusSchema,
  TaskType,
  TaskTypeSchema,
  TelemetryTarget,
  TerminalQuotaError,
  ToolCallDecision,
  ToolCallEvent,
  ToolConfirmationOutcome,
  ToolErrorType,
  ToolExecutionPhase,
  ToolExecutor,
  ToolOutputTruncatedEvent,
  ToolRegistry,
  TrackerAddDependencyTool,
  TrackerCreateTaskTool,
  TrackerGetTaskTool,
  TrackerListTasksTool,
  TrackerService,
  TrackerTaskSchema,
  TrackerUpdateTaskTool,
  TrackerVisualizeTool,
  Turn,
  UPDATE_TOPIC_DISPLAY_NAME,
  UPDATE_TOPIC_TOOL_NAME,
  UPGRADE_URL_PAGE,
  USER_POLICY_TIER,
  USER_STEERING_INSTRUCTION,
  UiTelemetryService,
  UnauthorizedError,
  UserAccountManager,
  UserPromptEvent,
  UserTierId,
  VALID_GEMINI_MODELS,
  ValidationCancelledError,
  ValidationRequiredError,
  ValueType,
  WEB_FETCH_DISPLAY_NAME,
  WEB_FETCH_PARAM_PROMPT,
  WEB_FETCH_TOOL_NAME,
  WEB_SEARCH_DISPLAY_NAME,
  WEB_SEARCH_PARAM_QUERY,
  WEB_SEARCH_TOOL_NAME,
  WORKSPACE_POLICY_TIER,
  WRITE_FILE_DISPLAY_NAME,
  WRITE_FILE_PARAM_CONTENT,
  WRITE_FILE_TOOL_NAME,
  WRITE_TODOS_TOOL_NAME,
  WarningPriority,
  WebFetchFallbackAttemptEvent,
  WebFetchTool,
  WebSearchTool,
  WindowsSandboxManager,
  WorkspaceContext,
  WorktreeService,
  WriteFileTool,
  WriteTodosTool,
  _clearCache,
  addMCPStatusChangeListener,
  addMemory,
  allowEditorTypeInSandbox,
  appendToLastTextPart,
  applyAdminAllowlist,
  applyReplacement,
  applyRequiredServers,
  assertValidPathString,
  assumeExhaustive,
  authEvents,
  buildG1Url,
  buildTodosReturnDisplay,
  buildToolResponseData,
  buildUserSteeringHintPrompt,
  bytesToMB,
  calculateReplacement,
  canUseRipgrep,
  categorizeAndConcatenate,
  checkExhaustive,
  checkPolicy,
  classifyGoogleError,
  cleanupWorktree,
  clearApiKey,
  clearCachedCredentialFile,
  clearEmittedPolicyWarnings,
  clearOauthClientCache,
  computeModelAddedAndRemovedLines,
  concatenateInstructions,
  connectAndDiscover,
  connectToMcpServer,
  contentPartsToGeminiParts,
  convertColorToHex,
  convertGithubUrlToRaw,
  convertSessionToClientHistory,
  convertToFunctionResponse,
  convertToRestPayload,
  coreEvents,
  createCache,
  createCodeAssistContentGenerator,
  createContentGenerator,
  createContentGeneratorConfig,
  createConversationOffered,
  createHookOutput,
  createPolicyEngineConfig,
  createPolicyUpdater,
  createSafeProxyAgent,
  createSandboxManager,
  createSessionId,
  createTranslationState,
  createTransport,
  createUnauthorizedToolError,
  createWorkingStdio,
  createWorktree,
  createWorktreeService,
  debugLogger,
  decodeTagName,
  deduplicatePathsByFileIdentity,
  deleteSessionArtifactsAsync,
  deleteSubagentSessionDirAndArtifactsAsync,
  detectBOM,
  detectEncodingFromBuffer,
  detectFileType,
  detectIdeFromEnv,
  detectLineEnding,
  disableBracketedPasteMode,
  disableKittyKeyboardProtocol,
  disableLineWrapping,
  disableModifyOtherKeys,
  disableMouseEvents,
  disableMouseTracking,
  discoverMcpTools,
  discoverPrompts,
  discoverResources,
  discoverTools,
  doesToolInvocationMatch,
  enableBracketedPasteMode,
  enableKittyKeyboardProtocol,
  enableLineWrapping,
  enableModifyOtherKeys,
  enableMouseEvents,
  encodeTagName,
  ensureRgPath,
  enterAlternateScreen,
  escapePath,
  escapeShellArg,
  estimateContextBreakdown,
  execStreaming,
  exitAlternateScreen,
  extractExtensionsFromPatterns,
  extractStringFromParseEntry,
  fetchAdminControls,
  fetchAdminControlsOnce,
  fetchWithTimeout,
  fileExists,
  filter,
  findGitRoot,
  findSecretFiles,
  flatMapTextParts,
  flattenMemory,
  flushTelemetry,
  formatBackgroundCompletionForModel,
  formatBytes,
  formatCheckpointDisplayList,
  formatForSpeech,
  formatMcpToolName,
  formatPolicyError,
  formatProtoJsonDuration,
  formatTruncatedToolOutput,
  formatUserHintsForModel,
  geminiPartsToContentParts,
  generalistProfile,
  generateCheckpointFileName,
  generateFastAckText,
  generateSteeringAckMessage,
  generateSummary,
  generateValidName,
  getActivityDetector,
  getActivityMonitor,
  getAdminBlockedMcpServersMessage,
  getAdminErrorMessage,
  getAgentCardLoadOptions,
  getAllGeminiMdFilenames,
  getAllMCPServerStatuses,
  getAlwaysAllowPriorityFraction,
  getApprovalModeDescription,
  getAuthTypeFromEnv,
  getAvailablePort,
  getCachedEncodingForBuffer,
  getCheckpointInfoList,
  getCitations,
  getCodeAssistServer,
  getCommandName,
  getCommandRoot,
  getCommandRoots,
  getCompatibilityWarnings,
  getCompressionPrompt,
  getConsentForOauth,
  getConventionAttributes,
  getCoreSystemPrompt,
  getCorrectedFileContent,
  getCurrentGeminiMdFilename,
  getDiffCommand,
  getDirectoryContextString,
  getDisplayString,
  getEditorCommand,
  getEditorDisplayName,
  getEnvironmentContext,
  getEnvironmentMemoryPaths,
  getErrorMessage,
  getErrorReplaceResult,
  getErrorStatus,
  getErrorType,
  getExperiments,
  getExtensionMemoryPaths,
  getFileDiffFromResultDisplay,
  getFolderStructure,
  getFsErrorMessage,
  getFunctionCalls,
  getFunctionCallsAsJson,
  getFunctionCallsFromParts,
  getFunctionCallsFromPartsAsJson,
  getG1CreditBalance,
  getGlobalMemoryFilePath,
  getGlobalMemoryPaths,
  getHookKey,
  getHookSource,
  getIdeInstaller,
  getInitialChatHistory,
  getMCPDiscoveryState,
  getMCPServerPrompts,
  getMCPServerStatus,
  getMemoryMonitor,
  getOauthClient,
  getPackageJson,
  getPathIdentity,
  getPlanModeExitMessage,
  getPolicyDenialError,
  getPolicyDirectories,
  getPolicyTier,
  getPreviousSession,
  getProjectHash,
  getProjectMemoryFilePath,
  getProjectRootForWorktree,
  getPromptIdWithFallback,
  getPty,
  getRealPath,
  getReleaseChannel,
  getRemoteAgentTargetUrl,
  getResponseText,
  getResponseTextFromParts,
  getRetryErrorType,
  getSecretFileFindArgs,
  getShellConfiguration,
  getSpecificMimeType,
  getStructuredResponse,
  getStructuredResponseFromParts,
  getSystemEncoding,
  getToolAliases,
  getToolCallDataSchema,
  getToolSuggestion,
  getTruncatedCheckpointNames,
  getUserProjectMemoryPaths,
  getVersion,
  getWorktreePath,
  handleFallback,
  hasCycleInSchema,
  hasNetworkTransport,
  hasRedirection,
  hasSummary,
  hasValidEditorCommand,
  hasValidEditorCommandAsync,
  hasWorktreeChanges,
  homedir,
  ideContextStore,
  initializeActivityMonitor,
  initializeMemoryMonitor,
  initializeShellParsers,
  initializeTelemetry,
  invokeMcpPrompt,
  isAbortError,
  isAccountSuspendedError,
  isActiveModel,
  isAddressPrivate,
  isApiError,
  isAppleTerminal,
  isAuthenticationError,
  isAutoModel,
  isBackgroundExecutionData,
  isBinary,
  isBinaryFile,
  isCloudShell,
  isCompletedAskUserTool,
  isCustomModel,
  isDumbTerminal,
  isEditToolParams,
  isEditorAvailable,
  isEditorAvailableAsync,
  isEmpty,
  isEnabled,
  isFatalToolError,
  isFileDiff,
  isGemini2Model,
  isGemini3Model,
  isGeminiWorktree,
  isGitRepository,
  isGnuScreen,
  isGrepResult,
  isGuiEditor,
  isHeadlessMode,
  isInvalidArgumentError,
  isJetBrainsTerminal,
  isListResult,
  isLoopbackHost,
  isLowColorTmux,
  isMcpToolAnnotation,
  isMcpToolName,
  isNightly,
  isNodeError,
  isOverageEligibleModel,
  isPerformanceMonitoringActive,
  isPreview,
  isPreviewModel,
  isPrivateIp,
  isPrivateIpAsync,
  isProModel,
  isReadManyFilesResult,
  isRecord,
  isRetryableError,
  isSchemaDepthError,
  isSecretFile,
  isStable,
  isStructuredError,
  isStructuredToolResult,
  isSubagentProgress,
  isSubpath,
  isTelemetrySdkInitialized,
  isTerminalEditor,
  isTmux,
  isTool,
  isToolActivityError,
  isToolCallResponseInfo,
  isUserActive,
  isUserVisibleHook,
  isValidNonThoughtTextPart,
  isValidPathString,
  isValidToolName,
  isWindows,
  isWindows10,
  isWithinRoot,
  isWriteFileToolParams,
  jsonToMarkdown,
  listExtensions,
  listMemoryFiles,
  loadAgentsFromDirectory,
  loadApiKey,
  loadExtensionPolicies,
  loadJitSubdirectoryMemory,
  loadPoliciesFromToml,
  loadServerHierarchicalMemory,
  loadSkillFromFile,
  loadSkillsFromDir,
  loadWasmBinary,
  logApiError,
  logApiRequest,
  logApiResponse,
  logBillingEvent,
  logChatCompression,
  logCliConfiguration,
  logConsecaPolicyGeneration,
  logConsecaVerdict,
  logConversationFinishedEvent,
  logExtensionDisable,
  logExtensionEnable,
  logExtensionInstallEvent,
  logExtensionUninstall,
  logExtensionUpdateEvent,
  logFlashFallback,
  logIdeConnection,
  logModelSlashCommand,
  logNetworkRetryAttempt,
  logOnboardingStart,
  logOnboardingSuccess,
  logRewind,
  logSlashCommand,
  logToolCall,
  logToolOutputTruncated,
  logUserPrompt,
  logWebFetchFallbackAttempt,
  makeChatCompressionEvent,
  makeFakeConfig,
  makeRelative,
  makeSlashCommandEvent,
  mapError,
  mapFinishReason,
  mapHttpToGrpcStatus,
  mapUsage,
  markdownToAgentDefinition,
  mcpServerRequiresOAuth,
  normalizeCommand,
  normalizePath,
  normalizeSpace,
  normalizeUrl,
  openBrowserSecurely,
  openDiff,
  parseAgentMarkdown,
  parseAndFormatApiError,
  parseBashCommandDetails,
  parseBooleanEnvFlag,
  parseCommandDetails,
  parseFrontmatter,
  parseGoogleApiError,
  parseMcpToolName,
  parsePrompt,
  parseTelemetryTargetValue,
  parseThought,
  partListUnionToString,
  partToString,
  patchStdio,
  performInit,
  performRestore,
  populateMcpServerCommand,
  processRestorableToolCalls,
  processSingleFileContent,
  promptIdContext,
  readFileWithEncoding,
  readGeminiMdFiles,
  readPathFromWorkspace,
  readPolicyFiles,
  readWasmBinaryFromDisk,
  recordApiErrorMetrics,
  recordApiRequestBreakdown,
  recordApiResponseMetrics,
  recordBaselineComparison,
  recordContentRetry,
  recordContentRetryFailure,
  recordConversationOffered,
  recordCpuUsage,
  recordCreditPurchaseClick,
  recordCurrentMemoryUsage,
  recordCustomApiResponseMetrics,
  recordCustomTokenUsageMetrics,
  recordExitFail,
  recordFileOperationMetric,
  recordFlickerFrame,
  recordGenAiClientOperationDuration,
  recordGenAiClientTokenUsage,
  recordInvalidChunk,
  recordMemoryUsage,
  recordModelRoutingMetrics,
  recordOverageOptionSelected,
  recordPerformanceRegression,
  recordPerformanceScore,
  recordRetryAttemptMetrics,
  recordSlowRender,
  recordStartupPerformance,
  recordTokenEfficiency,
  recordTokenUsageMetrics,
  recordToolCallInteractions,
  recordToolCallMetrics,
  recordToolExecutionBreakdown,
  recordToolQueueDepth,
  recordUserActivity,
  refreshMemory,
  refreshServerHierarchicalMemory,
  removeMCPStatusChangeListener,
  resetApiKeyCacheForTesting,
  resetBrowserSession,
  resetEncodingCache,
  resetOauthClientForTesting,
  resetUserDataCacheForTesting,
  resetVersionCache,
  resolveClassifierModel,
  resolveEditorAsync,
  resolveExecutable,
  resolveModel,
  resolvePathFromEnv,
  resolveSandboxPaths,
  resolveTelemetrySettings,
  resolveToRealPath,
  retryWithBackoff,
  runInDevTraceSpan,
  safeJsonToMarkdown,
  safeLiteralReplace,
  safeTemplateReplace,
  sanitizeAdminSettings,
  sanitizeFilenamePart,
  sanitizeHostname,
  sanitizeOutput,
  sanitizePaths,
  saveApiKey,
  saveTruncatedToolOutput,
  scheduleAgentTools,
  serializeTerminalToObject,
  sessionId,
  setGeminiMdFilename,
  setGlobalProxy,
  setupUser,
  shortenPath,
  shouldAttemptBrowserLaunch,
  shouldAutoUseCredits,
  shouldEnterAlternateScreen,
  shouldHideToolCall,
  shouldLaunchBrowser,
  shouldShowEmptyWalletMenu,
  shouldShowOverageMenu,
  showMemory,
  shutdownTelemetry,
  sortFileEntries,
  spawnAsync,
  splitCommands,
  startGlobalActivityMonitoring,
  startGlobalMemoryMonitoring,
  startMemoryService,
  startupProfiler,
  stopAdminControlsPolling,
  stopGlobalActivityMonitoring,
  stopGlobalMemoryMonitoring,
  stripShellWrapper,
  supports256Colors,
  supportsModernFeatures,
  supportsMultimodalFunctionResponse,
  supportsTrueColor,
  tildeifyPath,
  tmpdir,
  toFriendlyError,
  tokenLimit,
  toolResultDisplayToContentParts,
  translateEvent,
  truncateFastAckInput,
  truncateString,
  tryRealpath,
  uiTelemetryService,
  unescapePath,
  updateGlobalFetchTimeouts,
  updateMCPServerStatus,
  updatePolicy,
  validateAndSanitizeSessionId,
  validateMcpPolicyToolNames,
  validatePlanContent,
  validatePlanPath,
  windowsCodePageToEncoding,
  wrapInAccountChooser,
  writeToStderr,
  writeToStdout
};
